server/src/main/java/org/opensearch/action/search/SearchRequest.java

```diff
@@ -712,7 +713,13 @@ public final String buildDescription() {
             sb.append("scroll[").append(scroll.keepAlive()).append("], ");
         }
         if (source != null) {
-            sb.append("source[").append(source.toString(FORMAT_PARAMS)).append("]");
+            sb.append("source[");
+            try {
+                sb.append(source.toString(FORMAT_PARAMS));
+            } catch (final OpenSearchException ex) {
+                sb.append(ex.getMessage());
```

Comment:
I think we should not include the exception message as is, may be we could say:
```
  `sb.append("<exception thrown>");` 
```
Or

```
  `sb.append("n/a");` 
```

Just setting the source to error message gives an impression that this is the exact source, not a placeholder because of error

Reply:
The exception is useful IMO.

```
indices[], search_type[QUERY_THEN_FETCH], source[java.lang.UnsupportedOperationException: line ring cannot be serialized using GeoJson]
```

I don't want to die on this hill, what would you prefer?

Reply:
:D Fair enough, what about:  `sb.append("<error: " + ex.getMessage() +">");` @andrross wdyt?

Reply:
I like it, updated.

Reply:
`sb.append("<error: " + ex.getMessage() +">");` looks good to me